crates/forge/src/cmd/build.rs

```diff
@@ -228,6 +209,50 @@ impl BuildArgs {
             Ok([config.src, config.test, config.script, foundry_toml])
         })
     }
+
+    /// Check soldeer.lock file consistency using soldeer_core APIs
+    async fn check_soldeer_lock_consistency(&self, config: &Config) {
+        let soldeer_lock_path = config.root.join("soldeer.lock");
+        if !soldeer_lock_path.exists() {
+            return;
+        }
+
+        let lockfile = match soldeer_core::lock::read_lockfile(&soldeer_lock_path) {
+            Ok(lock) => lock,
+            Err(_) => return,
+        };
+
+        let deps_dir = config.root.join("dependencies");
+        for entry in &lockfile.entries {
+            let dep_name = entry.name();
+            let dep_path = deps_dir.join(format!("{}-{}", dep_name, entry.version()));
+
+            if !dep_path.exists() {
+                continue;
+            }
```

Comment:
This check is not necessary, as it's already performed by `check_dependency_integrity` which returns `DependencyStatus::Missing` if the folder is not there.